ow_lander/scripts/LanderInterface.py

```diff
@@ -19,69 +19,102 @@
+        # forward kinematics: enable forward kinematics service from moveit
+        rospy.wait_for_service('compute_fk')
+        try:
+            self.moveit_fk = rospy.ServiceProxy('compute_fk', GetPositionFK)
+        except rospy.ServiceException as e:
+            rospy.logerror("Service call failed: %s" % e)
+
+
 class LinkStateSubscriber:
-    
-  def __init__(self):
-    self._buffer = tf2_ros.Buffer()
-    self._listener = tf2_ros.TransformListener(self._buffer)
-    self._link_value = None
-    self.subscriber = rospy.Subscriber("/gazebo/link_states", LinkStates, self._handle_link_states)
-    
-    
-  def _handle_link_states(self, data):
-    # position of the end of scoop can be obtained either from raw Gazebo readings
-    # of Link states or from tf2. TF2 method is commented out . Link State method is 
-    # being used in the current implementation
-      
-    """
-    :type data: gazebo_msgs.msg._LinkStates.LinkStates
-    """
-    try:
-     idx = data.name.index("lander::l_scoop")
-    except ValueError:
-      rospy.logerr_throttle(
-          1, "LanderInterface: lander::l_scoop_tip not found in link_states")
-      return
-    self._link_value = (data.pose[idx].position)
-    
-    
-    #try:
-      #t = self._buffer.lookup_transform(
-          #"base_link", "l_scoop_tip", rospy.Time())
-    #except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
-      #rospy.logwarn("LanderInterface: tf2 raised an exception")
-      #return None
-    ##return t.transform.translation
-    #self._link_value = t.transform.translation
-  
+
+    def __init__(self):
+        self._buffer = tf2_ros.Buffer()
+        self._listener = tf2_ros.TransformListener(self._buffer)
+        self._link_value = None
+        self.subscriber = rospy.Subscriber(
+            "/gazebo/link_states", LinkStates, self._handle_link_states)
+
+    def _handle_link_states(self, data):
+        # position of the end of scoop can be obtained either from raw Gazebo readings
+        # of Link states or from tf2. TF2 method is commented out . Link State method is
+        # being used in the current implementation
+        """
+        :type data: gazebo_msgs.msg._LinkStates.LinkStates
+        """
+        try:
+            idx = data.name.index("lander::l_scoop")
+        except ValueError:
+            rospy.logerr_throttle(
+                1, "LanderInterface: lander::l_scoop_tip not found in link_states")
+            return
+        self._link_value = (data.pose[idx].position)
+
+        # try:
+        # t = self._buffer.lookup_transform(
+        # "base_link", "l_scoop_tip", rospy.Time())
+        # except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
+        #rospy.logwarn("LanderInterface: tf2 raised an exception")
+        # return None
+        # return t.transform.translation
+        #self._link_value = t.transform.translation
+
+
+class JointStateSubscriber:
```

Comment:
Since this only provides arm joints, I think the name should reflect that.
```suggestion
class ArmJointStateSubscriber:
```